nocopy.go

```diff
@@ -0,0 +1,24 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package concert
+
+// Types using with noCopy will be flagged by `go vet`.
+// See: https://github.com/golang/go/issues/8005#issuecomment-190753527
+type noCopy struct{}
```

Comment:
it's not required by RefCount anymore (due to the mutex). Still I'm leaving this here just in case we need it in the future.
If used as a field in a struct, then `go vet` will complain if a value of said type is copied.